pkg/venafi/tpp/connector.go

```diff
@@ -582,6 +582,21 @@ func prepareRequest(req *certificate.Request, zone string) (tppReq certificateRe
 		tppReq.CASpecificAttributes = append(tppReq.CASpecificAttributes, nameValuePair{Name: expirationDateAttribute, Value: formattedExpirationDate})
 	}
 
+	if req.ValidityDuration != nil {
+		for _, attributeName := range []string{
+			"Microsoft CA:Specific End Date",
+			"DigiCert CA:Specific End Date",
+			"EntrustNET CA:Specific End Date",
+			"Specific End Date",
+		} {
+			formattedExpirationDate := time.Now().Add(*req.ValidityDuration).Format(time.RFC3339)
```

Comment:
TPP supports parsing multiple datetime formats, this RFC3339 also includes the timezone and thus limits the risk of having a timezone shift (I tested that this results in the correct End Date for the MS CA).

Reply:
Does the timezone aware date string get passed to the issuer? 
Do we know that all the issuers support datetimes with timezone info?
The original code (above) does not seem to add TZ info to the formatted, and I wonder if there's a reason for that? 
It does take the trouble to compute a UTC date, and I wonder if there's a reason for that?

Could the reason be that the various issuers expect non-timezone aware UTC datetime string?

Maybe check with Ryan Treat or who ever wrote that existing code.

Reply:
`Does the timezone aware date string get passed to the issuer?`
- based on what I read in the TPP source code, it seems to be parsed by TPP before passing the value to the issuer.